lib/features/base/base_controller.dart

```diff
@@ -119,71 +156,107 @@ abstract class BaseController extends GetxController
     if (exception is ConnectionError) {
-      if (currentOverlayContext != null && currentContext != null) {
-        appToast.showToastErrorMessage(
-          currentOverlayContext!,
-          AppLocalizations.of(currentContext!).connectionError
-        );
-      }
+      _handleConnectionErrorException();
+    } else if (exception is BadCredentialsException) {
+      _handleBadCredentialsException();
     }
+  }
 
-    if (!authorizationInterceptors.isAppRunning) {
-      return;
+  void handleUrgentExceptionOnWeb({Failure? failure, Exception? exception}) {
+    logError('$runtimeType::handleUrgentExceptionOnWeb():Failure: $failure | Exception: $exception');
+    if (exception is NoNetworkError) {
+      _handleNotNetworkErrorException();
+    } else if (exception is ConnectionError) {
+      _handleConnectionErrorException();
+    } else if (exception is BadCredentialsException) {
+      _handleBadCredentialsException();
     }
-    _executeBeforeUnloadAndLogOut(exception);
   }
 
-  Future<void> _executeBeforeUnloadAndLogOut(Exception? exception) async {
-    if (exception is BadCredentialsException || exception is ConnectionError) {
-      if (PlatformInfo.isWeb) {
-        await executeBeforeUnload();
+  Future<void> _executeBeforeUnloadAndLogOut() async {
+    await executeBeforeUnload();
+    clearDataAndGoToLoginPage();
+  }
+
+  void _handleConnectionErrorException() {
+    if (currentOverlayContext != null && currentContext != null) {
+      appToast.showToastErrorMessage(
+        currentOverlayContext!,
+        AppLocalizations.of(currentContext!).connectionError);
+    }
+  }
+
+  void _handleNotNetworkErrorException() {
+    if (currentOverlayContext != null && currentContext != null) {
+      appToast.showToastMessage(
+        currentOverlayContext!,
+        AppLocalizations.of(currentContext!).no_internet_connection,
+        actionName: AppLocalizations.of(currentContext!).skip,
+        onActionClick: ToastView.dismiss,
+        leadingSVGIcon: imagePaths.icNotConnection,
+        backgroundColor: AppColor.textFieldErrorBorderColor,
+        textColor: Colors.white,
+        infinityToast: true);
+    }
+  }
+
+  void _handleBadCredentialsException() {
+    if (PlatformInfo.isWeb) {
+      if (currentContext == null) {
+        _executeBeforeUnloadAndLogOut();
+        return;
       }
+
+      showConfirmDialogAction(
+        currentContext!,
+        AppLocalizations.of(currentContext!).dialogMessageSessionHasExpired,
+        AppLocalizations.of(currentContext!).reconnect,
+        title: AppLocalizations.of(currentContext!).sessionExpired,
+        alignCenter: true,
+        outsideDismissible: false,
+        icon: SvgPicture.asset(imagePaths.icTMailLogo, width: 64, height: 64),
+        paddingButton: const EdgeInsets.symmetric(horizontal: 8, vertical: 16),
+        onConfirmAction: _executeBeforeUnloadAndLogOut);
```

Comment:
because designer need the dialog consistent, can we use `browser` dialog here?

Reply:
Done